### ISSUE TITLE
md-autocomplete: Loading bar remains visible after item has been selected

### ISSUE THREAD

**Issue by diogodomanski** [angular/material#3563]
I already wasted many hours trying to solve this problem, so I decided to ask for help here.

I'm using `<md-autocomplete>` inside a `<form>` as follows:

```html
<form layout="column" name="myForm">
<md-autocomplete
						md-input-name="locale"
						md-no-cache="true"
						md-selected-item="obj.user.locale"
						md-search-text-change="null"
						md-search-text="localeSearch"
						md-selected-item-change="null"
						md-items="item in listLocales(localeSearch)"
						md-item-text="item.name"
						md-min-length="0"
						md-delay="500"
						md-floating-label="Locale"
						required>
						<md-item-template>
							<span md-highlight-text="localeSearch" md-highlight-flags="^i">{{item.name}}</span>
						</md-item-template>
						<md-not-found>
							No matches found for {{localeSearch}}.
						</md-not-found>
						<div ng-messages="myForm.locale.$error">
							<div ng-message="required">This is required</div>
						</div>
					</md-autocomplete>
</form>
```

My controller has the following listLocales() method:

```javascript
$scope.listLocales = function (term) {
			return myService.listLocales(term).then(
					function (response) {
						if (response == undefined)
							return [];

						if (response.code == 0) {

							return $scope.locales = response.data;

						} else {
							toastService.pop(response.message, STATUS_ERROR);
						}
					}, function (error) {
				if (error.message == "")
					return;
				toastService.pop(error.message, STATUS_ERROR);
			}, function (notify) {
				console.log(notify);
			}
			);
		};
```

And, finally, in myService (invoked from controller) is the following:

```javascript
this.listLocales = function (term) {
					if (this.listLocalesPromise)
						this.listLocalesPromise.abort();

					var defer = $q.defer();

					var request = $http({
						method: 'GET',
						url: baseUrl + 'api/list-locales?q=' + term,
						timeout: defer.promise
					});

					this.listLocalesPromise = request.then(
							function (response) {
								return $q.resolve(response.data);
							},
							function (response) {
								return $q.reject(crudService.prepareError(response));
							}
					);

					this.listLocalesPromise.abort = function () {
						defer.resolve();
					};

					this.listLocalesPromise.finally(
							function () {
								this.listLocalesPromise.abort = angular.noop;
								defer = this.listLocalesPromise = null;
							}
					);

					return this.listLocalesPromise;
				};
```
After typing some text in md-autocomplete element and selecting an item, two undesirable things are happening:
1. The lisLocales() method is being invoked again (because, by selecting an item, the value of localeSearch is being changed)
2. The progress bar is being shown (and does not hide anymore)

Did anyone faced these issues? Are they bugs or am I doing something wrong?

Thanks

Diogo

**Comment by ThomasBurleson** [comment]
@diogodomanski -
Please post this in the [Angular Material Forum](https://groups.google.com/forum/#!forum/ngmaterial).
Note that this markup is used in AutoComplete:
```html
 <md-progress-linear
              ng-if="$mdAutocompleteCtrl.loading && !$mdAutocompleteCtrl.hidden"
              md-mode="indeterminate">
</md-progress-linear>
```

**Comment by mzbyszynski** [comment]
I think it's the `md-min-length="0"` that causes this.

**Comment by ThomasBurleson** [comment]
@mzbyszynski - still sounds like a configuration bug.

**Comment by diogodomanski** [comment]
@ThomasBurleson here is CodePen url: http://codepen.io/anon/pen/QbQBXQ. 

I've cleared the code to keep only what is important.

**Comment by robertmesserle** [comment]
@ThomasBurleson I believe this is a duplicate of an issue that was resolved post-0.10.0.

Confirmed here: http://codepen.io/robertmesserle/pen/MwQqVd

Closing.
